### PR TITLE
Code and comments details

### PR DESCRIPTION
**Pull request by hugocbpassos**
Check the reviews down below to understand the changes.

### CODE REVIEW

File: lib/audioplayers.dart [hunk 1]
```diff
@@ -161,8 +159,7 @@ class AudioPlayer {
         null, // Must be null by default to be compatible with radio streams
     bool respectSilence: false,
   }) async {
-    final int positionInMilliseconds =
-        position == null ? null : position.inMilliseconds;
+    final int positionInMilliseconds = position?.inMilliseconds;
```
Comment (hugocbpassos) [inline lib/audioplayers.dart]: If `position` is `null`, `position?` will return `null` before calling `inMilliseconds`.

File: lib/audioplayers.dart [hunk 2]
```diff
@@ -87,9 +87,7 @@ class AudioPlayer {
 
   set state(AudioPlayerState state) {
     _playerStateController.add(state);
-    if (audioPlayerStateChangeHandler != null) {
-      audioPlayerStateChangeHandler(state);
-    }
+    audioPlayerStateChangeHandler?.call(state);
```
Comment (hugocbpassos) [inline lib/audioplayers.dart]: Not a very popular syntax, but does the work gracefully.

File: lib/audio_cache.dart [hunk 3]
```diff
@@ -29,24 +29,24 @@ class AudioCache {
...
   /// This flag should be set to true, if player is used for playing internal notifications
   ///
-  /// This flag will have influence of stream type. And will respect silent mode if set to true
+  /// This flag will have influence of stream type, and will respect silent mode if set to true
   bool respectSilence;
 
   AudioCache({this.prefix = "", this.fixedPlayer, this.respectSilence = false});
 
-  /// Clear the cache of the file [fileName].
+  /// Clears the cache of the file [fileName].
```
Comment (hugocbpassos) [inline lib/audio_cache.dart]: You are going to see a lot of these changes down below. Both "clear" and "clears" are correct, but you got to set and follow a pattern.

File: lib/audio_cache.dart [hunk 4]
```diff
@@ -9,7 +9,7 @@ import 'audioplayers.dart';
 
 /// This class represents a cache for Local Assets to be played.
 ///
-/// Flutter can only play audios on device folders, so this first copies files to a temporary folder and the plays then.
+/// Flutter can only play audios on device folders, so this first one copies files to a temporary folder and then plays them.
```
Comment (luanpotter) [inline lib/audio_cache.dart]: Regarding just the second change on this phrase, I think you misunderstood, it was supposed to be

```
Flutter can only play audios on device folders, so this class first copies the files to a temporary folder and then plays them.
```
The other fixes are perfect!

File: lib/audio_cache.dart (hunk 3 above)
Comment (luanpotter) [reply lib/audio_cache.dart]: Agreed, we can keep the third person throughout.

File: lib/audio_cache.dart [hunk 5]
```diff
@@ -29,24 +29,24 @@ class AudioCache {
...
+  /// This flag will have influence of stream type, and will respect silent mode if set to true
   bool respectSilence;
 
   AudioCache({this.prefix = "", this.fixedPlayer, this.respectSilence = false});
 
-  /// Clear the cache of the file [fileName].
+  /// Clears the cache of the file [fileName].
   ///
-  /// Does nothing if there was already no cache.
+  /// Does nothing if there was no cache at all.
```
Comment (luanpotter) [inline lib/audio_cache.dart]: This phrase is still weird like that. Maybe:

```
    Does nothing if the file was not on the cache.
```

File: lib/audioplayers.dart (hunk 2 above)
Comment (luanpotter) [reply lib/audioplayers.dart]: More than gracefully, I'd say, it's at least ten times prettier! Nice changes (all occurrences) ;)

File: lib/audio_cache.dart (hunk 4 above)
Comment (hugocbpassos) [reply lib/audio_cache.dart]: Oh, ok. I'll change that. Thanks for warning me.

File: lib/audio_cache.dart (hunk 5 above)
Comment (hugocbpassos) [reply lib/audio_cache.dart]: The more clear, the better. I'll keep your suggestion.

**Comment by hugocbpassos** [comment]
Please, take a look at the new changes.

**Comment by luanpotter** [comment]
Perfect!

**Pull request merged**